src/express.re

```diff
@@ -1,39 +1,147 @@
-  /** [params request] return the JSON object filled with the 
+  /** [params request] return the JSON object filled with the
       request parameters */
 
   external asJsonObject : t => Js_dict.t Js_json.t = "%identity";
-  /** [asJsonObject request] casts a [request] to a JSON object. It is 
-      common in Express application to use the Request object as a 
-      placeholder to maintain state through the various middleware which 
-      are executed. */ 
-
-  /** more property method coming later */
+  /** [asJsonObject request] casts a [request] to a JSON object. It is
+      common in Express application to use the Request object as a
+      placeholder to maintain state through the various middleware which
+      are executed. */
+
+  external baseUrl: t => string = "" [@@bs.get];
+  /** [baseUrl request] returns the 'baseUrl' property */
+
+  external hostname : t => string = "" [@@bs.get];
+  /** [hostname request] Contains the hostname derived from the Host
+      HTTP header.*/
+
+  external ip : t => string = "" [@@bs.get];
+  /** [ip request] Contains the remote IP address of the request.*/
+
+  external fresh : t => bool =  "" [@@bs.get];
+  /** [fresh request] returns [true] whether the request is "fresh" */
+
+  external stale : t => bool =  "" [@@bs.get];
+  /** [stale request] returns [true] whether the request is "stale"*/
+
+  external methodRaw : t => string = "method" [@@bs.get];
+
+  type method_ =
+    | Get
+    | Post
+    | Put
+    | Delete
+    | Head
+    | Options
+    | Trace
+    | Connect;
+
+  let method_ : t => method_ = fun req => {
+    switch (methodRaw req) {
+      | "GET" => Get;
+      | "POST" => Post;
+      | "PUT" => Put;
+      | "DELETE" => Delete;
+      | "HEAD" => Head;
+      | "OPTIONS" => Options;
+      | "TRACE" => Trace;
+      | "CONNECT" => Connect;
+      | s => failwith ("Express.Request.method_ Unexpected method: " ^ s);
+    }};
+  /** [method_ request] return a string corresponding to the HTTP
+      method of the request: GET, POST, PUT, and so on */
+
+  external originalUrl : t => string = "" [@@bs.get];
+  /** [originalUrl request] returns the original url. See
+      https://expressjs.com/en/4x/api.html#req.originalUrl */
+
+  external path : t => string = "" [@@bs.get];
+  /** [path request] returns the path part of the request URL.*/
+
+  external protocolRaw : t => string = "protocol" [@@bs.get];
+
+  type protocol =
+    | Http
+    | Https;
+
+  let protocol : t => protocol = fun req => {
+    switch (protocolRaw req) {
+      | "http" => Http;
+      | "https" => Https;
+      | s => failwith ("Express.Request.protocol, Unexpected protocol: " ^ s)
+    }};
+  /** [protocol request] returns the request protocol string: either http
+      or (for TLS requests) https. */
+
+  external secure : t => bool = "" [@@bs.get];
+  /** [secure request] returns [true] if a TLS connection is established */
+
+  external query : t => Js.Dict.t Js.Json.t = "" [@@bs.get];
```

Comment:
it's a nested json object, since a query can be arbitrarily nested. express defaults to using https://www.npmjs.com/package/qs as its query string parser, but you can also supply your own. [description](https://expressjs.com/en/api.html#app.set)